### PR TITLE
Translate the remaining test vector structs to JSON

### PR DESCRIPTION
**Pull request by bifurcation**
Finishing what we started in #33.

### CODE REVIEW

File: test-vectors.md [hunk 1]
```diff
@@ -152,43 +152,46 @@ Parameters:
...
+      "commit_secret": /* hex-encoded binary data */,
+      "psk_secret": /* hex-encoded binary data */,
+      
+      "confirmed_transcript_hash": /* hex-encoded binary data */,
+      "interim_transcript_hash": /* hex-encoded binary data */,
+      "group_context": /* hex-encoded binary data */,
+      
+      "joiner_secret": /* hex-encoded binary data */,
+      "welcome_secret": /* hex-encoded binary data */,
+      "epoch_secret": /* hex-encoded binary data */,
```
Comment (franziskuskiefer) [inline test-vectors.md:177]: The epoch secret is a temporary value in the key schedule that should never get exposed. We can keep it in here but it requires extra code for testing and should always be tested through all the secrets that follow here. I'd remove it here.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Generally looks good to me with one change I'd suggest on the key schedule test vector.

The others look good to me. But I haven't implemented them yet.

File: test-vectors.md [hunk 2]
```diff
@@ -152,43 +152,46 @@ Parameters:
...
-  CipherSuite cipher_suite;
-  CryptoValue group_id;
-  CryptoValue initial_tree_hash;    // chosen by generator
-  CryptoValue initial_init_secret;  // chosen by generator
-  Epoch epochs<0..2^32-1>;
-} KeyScheduleTestVector;
+{
+  "cipher_suite": /* uint16 */,
+  "group_id": /* hex-encoded binary data */,
+  "initial_tree_hash": /* hex-encoded binary data */,
```
Comment (franziskuskiefer) [inline test-vectors.md:158]: With the `tree_hash` for each epoch I'm not sure we need this one.

File: test-vectors.md [hunk 3]
```diff
@@ -152,43 +152,46 @@ Parameters:
...
+  "initial_init_secret": /* hex-encoded binary data */,
+  "epochs": [
+    {
+      // These fields are chosen by the generator.  The "commit" field contains
+      // TLS-serialized MLSPlaintext object.  The membership_tag and
+      // confirmation_tag in the MLSPlaintext MUST be valid; otherwise its
+      // content is only used for the transcript.
+      "commit": /* hex-encoded binary data */,
```
Comment (franziskuskiefer) [inline test-vectors.md:166]: To only test the key schedule we don't actually need a commit. I would suggest to leave the message out of this test vector.
So the beginning here would be
```
// Chosen by generator
"tree_hash": /* hex-encoded binary data */,
"commit_secret": /* hex-encoded binary data */,
"psk_secret": /* hex-encoded binary data */,
"confirmed_transcript_hash": /* hex-encoded binary data */,

// Computed values
"group_context": /* hex-encoded binary data */,
...
```

Comment (bifurcation) [reply test-vectors.md:166]: Having the commit also lets us test transcript hashing, and the confirmation tag / membership tag logic, which we don't get if we don't have the commit.  Would you prefer to get that coverage here or in a separate test vector?

Comment (bifurcation) [reply test-vectors.md:166]: To be clear, the alternative would be to add a "Commit and Transcript" test vector something like:

```text
{
  // Chosen by the generator
  "interim_transcript_hash_before": /* hex-encoded binary data */,
  "membership_key": /* hex-encoded binary data */,
  "confirmation_key": /* hex-encoded binary data */,
  "commit": /* hex-encoded TLS-serialized MLSPlaintext(Commit) */

  "confirmed_transcript_hash_after": /* hex-encoded binary data */,
  "interim_transcript_hash_after": /* hex-encoded binary data */,
}
```

Verification:
* Verify that `confirmed_transcript_hash_after` is the result of updating `interim_transcript_hash_before` with `commit`
* Verify that `commit.confirmation_tag` is present and verifies using `confirmed_transcript_hash_after` and `confirmation_key`
* Verify that `commit.membership_tag` is present and verifies using `membership_key`

Comment (franziskuskiefer) [reply test-vectors.md:166]: Yeah I guess it depends here on what exactly we want to test with this test vector. I had the key schedule in mind only and the hashes here test transcript/group state. So this felt a little out of place. I think separate test vectors would be cleaner. But I'm fine either way.

Comment (bifurcation) [reply test-vectors.md:166]: I can agree with that.  I'll split it out.

**Pull request merged**